### ISSUE TITLE
Modernize Models & Validation with Pydantic

### ISSUE THREAD

**Issue by ax3l** [picmi-standard/picmi#94]
It was suggested today by @roussel-ryan to use a more formal schema to abstract PICMI inputs and extensions and compatibility with different codes.

Refs.:
- https://docs.pydantic.dev/usage/models/
- https://docs.pydantic.dev/usage/validators/

**Comment by roussel-ryan** [comment]
Reference for use in xopt https://github.com/ChristopherMayes/Xopt/blob/main/xopt/generators/bayesian/options.py

**Comment by ax3l** [comment]
This is similar to   #63 (more) / #71 (less) but with clear paths to validate and being able to export data models for post-processing.

**Comment by BrianMarre** [comment]
I agree with @ax3l abstracting PICMI further from the codes would be a good idea
and testing for correct configuration on the picmi side would also be helpfull.
